File: test/Integration/test-java/src/main/java/com/function/AddProductsNoPartialUpsert.java
package com.function;

import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.OutputBinding;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.sql.annotation.SQLOutput;
import com.function.Common.Product;

import java.util.Optional;

public class AddProductsNoPartialUpsert {
    @FunctionName("AddProductsNoPartialUpsert")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.POST},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "addproduct-nopartialupsert")
                HttpRequestMessage<Optional<String>> request,
            @SQLOutput(
                name = "products",
                commandText = "dbo.ProductsNameNotNull",
                connectionStringSetting = "SqlConnectionString")
                OutputBinding<Product[]> products) {

        Product validProduct = new Product(0, "test", 100);
        Product invalidProduct = new Product(1, null, 100);
        Product[] p = new Product[]{validProduct, invalidProduct};
        products.setValue(p);

        return request.createResponseBuilder(HttpStatus.OK).header("Content-Type", "application/json").body(products).build();
    }
}

File: test-outofproc/AddProductsNoPartialUpsert.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using Microsoft.Azure.Functions.Worker.Extensions.Sql;
using Microsoft.Azure.Functions.Worker;
using Microsoft.AspNetCore.Http;
using System.Collections.Generic;
using DotnetIsolatedTests.Common;

namespace DotnetIsolatedTests
{
    public static class AddProductsNoPartialUpsert
    {
        public const int UpsertBatchSize = 1000;

        // This output binding should throw an error since the ProductsNameNotNull table does not 
        // allows rows without a Name value. No rows should be upserted to the Sql table.
        [Function("AddProductsNoPartialUpsert")]
        [SqlOutput("dbo.ProductsNameNotNull", "SqlConnectionString")]
        public static List<Product> Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "addproducts-nopartialupsert")]
            HttpRequest req)
        {
            List<Product> newProducts = ProductUtilities.GetNewProducts(UpsertBatchSize);
            foreach (Product product in newProducts)
            {
                newProducts.Add(product);
            }

            var invalidProduct = new Product
            {
                Name = null,
                ProductId = UpsertBatchSize,
                Cost = 100
            };
            newProducts.Add(invalidProduct);

            return newProducts;
        }
    }
}
